test/integration_test.py

```diff
@@ -386,13 +386,25 @@ def test_catalog_listing(self):
 
     def test_snapshot_listing(self):
         """
-        Test with a small page size to be sure paging works
+        Test with small page sizes to make sure paging works.
         """
-        page_size = 5
-        with mock.patch.object(TDRClient, 'page_size', page_size):
-            paged_snapshots = self._public_tdr_client.snapshot_names_by_id()
-        snapshots = self._public_tdr_client.snapshot_names_by_id()
-        self.assertEqual(snapshots, paged_snapshots)
+        # Without a filter, the test takes so long that there's a real risk of
+        # failure due to new snapshots being added mid-test.
+        snapshot_filters_by_deployment = {
+            'anvildev': 'anvil_',  # ~5 snapshots
+            'dev': 'hca_dev_5',  # ~10 snapshots
+            'anvilprod': 'anvil_page_',  # ~13 snapshots
+            'prod': '_dcp37'  # ~13 snapshots
+        }
+        filter = snapshot_filters_by_deployment[config.main_deployment_stage]
+        for page_size in 1, 2:
+            with self.subTest(page_size=page_size):
+                with mock.patch.object(TDRClient, 'page_size', page_size):
+                    paged_snapshots = self._tdr_client.snapshot_names_by_id(filter=filter)
+                    self.assertGreater(len(paged_snapshots), 0)
+                snapshots = self._tdr_client.snapshot_names_by_id(filter=filter)
+                self.assertLess(len(snapshots), 20)
```

Comment:
Since we assert that the lists are equal, it's a bit confusing to make separate assertions about their size. I believe this simplifies to:

```suggestion
                snapshots = self._tdr_client.snapshot_names_by_id(filter=filter)
                self.assertIn(len(snapshots), range(1, 20))
```